rag/utils/gaussdb_conn.py

```
@@ -0,0 +1,1431 @@
+            knowledgebase_id,
+            is_meta=False,
+        )
+        if not where_sql:
+            return False
+
+        conn = self.pool.get_conn()
+        cur = None
+        try:
+            cur = conn.cursor()
+            cur.execute(
+                f"SELECT id, kb_id, extra FROM {table} WHERE {where_sql} FOR UPDATE",
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)